src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala

```diff
@@ -35,17 +35,24 @@ class TaskOpFactoryHelper(
     TaskOp.Launch(taskInfo, newTask, Some(oldTask), createOperations)
   }
 
+  /**
+    * Returns a set of operations to reserve ALL resources (cpu, mem, ports, disk, etc.) and then create persistent
+    * volumes against them as needed
+    */
   def reserveAndCreateVolumes(
     frameworkId: FrameworkId,
     newTask: TaskStateOp.Reserve,
     resources: Iterable[Mesos.Resource],
-    localVolumes: Iterable[LocalVolume],
+    localVolumes: Iterable[(Option[Mesos.Resource.DiskInfo.Source], LocalVolume)],
```

Comment:
Actually, there is a good reason to use `Option[Mesos.Resource.DiskInfo.Source]` here and not `DiskType`, as `DiskType` does not indicate the corresponding path for the resource.

I could change it to `Iterable[(DiskType, Option[String], LocalVolume)` or something... or create a new type? `Allocation` or something, which contains all of the signal necessary to create the appropriate mesos resource allocation request?